matrix_sdk_common/src/deserialized_responses.rs

```diff
@@ -37,6 +43,72 @@ pub struct AmbiguityChanges {
     pub changes: BTreeMap<RoomId, BTreeMap<EventId, AmbiguityChange>>,
 }
 
+/// The verification state of the device that sent an event to us.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum VerificationState {
+    /// The device is trusted.
+    Trusted,
+    /// The device is not trusted.
+    Untrusted,
+    /// The device is not known to us.
+    UnknownDevice,
+}
+
+/// The algorithm specific information of a decrypted event.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum AlgorithmInfo {
+    /// The info if the event was encrypted using m.megolm.v1.aes-sha2
+    MegolmV1AesSha2 {
+        /// The curve25519 key of the device that created the megolm decryption
+        /// key originally.
+        curve25519_key: String,
+        /// The signing keys that have created the megolm key that was used to
+        /// decrypt this session. This map will usually contain a signle ed25519
+        /// key.
+        sender_claimed_keys: BTreeMap<DeviceKeyAlgorithm, String>,
+        /// Chain of curve25519 keys through which this session was forwarded,
+        /// via m.forwarded_room_key events.
+        forwarding_curve25519_key_chain: Vec<String>,
+    },
+}
+
+/// Struct containing information on how an event was decrypted.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct EncryptionInfo {
+    /// The user ID of the event sender, note this is untrusted data unless the
+    /// `verification_state` is as well trusted.
+    pub sender: UserId,
+    /// The device ID of the device that sent us the event, note this is
+    /// untrusted data unless `verification_state` is as well trusted.
+    pub sender_device: DeviceIdBox,
+    /// Information about the algorithm that was used to encrypt the event.
+    pub algorithm_info: AlgorithmInfo,
+    /// The verification state of the device that sent us the event, note this
+    /// is the state of the device at the time of decryption. It may change in
+    /// the future if a device gets verified or deleted.
+    pub verification_state: VerificationState,
+}
+
+/// A customized version of a room event comming from a sync that holds optional
+/// decryption info.
```

Comment:
```suggestion
/// encryption info.
```

If the type is called that, the docs should use the same term, no? ^^